src/main/app/claims/responseModelConverter.ts

```diff
@@ -1,40 +1,57 @@
 import { ResponseDraft } from 'response/draft/responseDraft'
+import { ResponseData } from 'response/draft/responseData'
+import { PartyType } from 'forms/models/partyType'
+import { IndividualDetails } from 'forms/models/individualDetails'
+import { Party } from 'app/claims/models/details/yours/party'
+import { Individual } from 'app/claims/models/details/yours/individual'
+import { Company } from 'app/claims/models/details/yours/company'
+import { Address } from 'app/claims/models/address'
+import { Organisation } from 'app/claims/models/details/yours/organisation'
+import { SoleTrader } from 'app/claims/models/details/yours/soleTrader'
 
 export class ResponseModelConverter {
 
-  static convert (responseDraft: ResponseDraft): any {
-    responseDraft.defendantDetails.mobilePhone = responseDraft.defendantDetails.mobilePhone.number as any
-    responseDraft.defendantDetails.dateOfBirth = responseDraft.defendantDetails.dateOfBirth.date.asString() as any
+  static convert (responseDraft: ResponseDraft): ResponseData {
 
-    this.convertPartyDetails(responseDraft)
-
-    const response = {
-      response: responseDraft.response.type.value as string,
-      defendant: responseDraft.defendantDetails
-    }
-
-    if (responseDraft.requireDefence()) {
-      response['defence'] = responseDraft.defence.text as string
-      response['freeMediation'] = responseDraft.freeMediation.option as string
-    }
-
-    return response
+    const responseData = new ResponseData(responseDraft.response.type.value,
+                                          responseDraft.defence.text,
+                                          responseDraft.freeMediation == null ? undefined : responseDraft.freeMediation.option,
+                                          responseDraft.moreTimeNeeded == null ? undefined : responseDraft.moreTimeNeeded.option,
+                                          this.convertPartyDetails(responseDraft))
+    return responseData
   }
 
-  private static convertPartyDetails (responseDraft: ResponseDraft): void {
-    responseDraft.defendantDetails['address'] = responseDraft.defendantDetails.partyDetails.address
+  private static convertPartyDetails (responseDraft: ResponseDraft): Party {
```

Comment:
Seems like a lot of duplicated code. The same logic is in the claim converter. It does not make sense to have 2 deserialisation logic, 40 lines each to deserialise the same object.

ACTION NEEDED